community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StoreAbstractRelationshipCursor.java

```diff
@@ -107,15 +114,28 @@ public long otherNode( long nodeId )
                 relationshipRecord.getSecondNode() : relationshipRecord.getFirstNode();
     }
 
+    private Lock shortLivedReadLock()
+    {
+        Lock lock = lockService.acquireRelationshipLock( relationshipRecord.getId(), LockService.LockType.READ_LOCK );
+        if ( lockService != NO_LOCK_SERVICE )
+        {
+            // It's safer to re-read the relationship record here, specifically nextProp, after acquiring the lock
+            relationshipStore.fillRecord( relationshipRecord.getId(), relationshipRecord, FORCE );
```

Comment:
Maybe it's better to use `RecordLoad.CHECK` and check if it returns `false`, in which case the record has been deleted?